### PR TITLE
Add a spellcheck yarn job

### LINKED ISSUE #115: Provide spell checker process
Currently there are not spell checks run as part of a linter or other process.  That allows mispelled words in MDX and example files.  The `cspell` tool in node.js is helpful to find these.  Having a tool that could output warning of possible misspellings would be helpful.   

`npx cspell **/*.md* **/*.yaml`

### PR DESCRIPTION
**Pull request by ptgott**
Closes #115

Add a `spellcheck` yarn script that we can run in the `gravitational/teleport` CI job for linting docs changes.

This change assumes that there is a `cspell.json` configuration file at the root of each supported docs version that includes words to ignore during the spellcheck.

Note that this command uses `find` with `xargs` instead of `find` with the `-exec` flag because the former returns a nonzero exit code if one of the `-exec` runs fails.

### CODE REVIEW

**Comment by ptgott** [comment]
CC @zmb3 @stevenGravy

File: package.json
Comment (zmb3) [inline package.json:7]: Can we do without the `find`? Is yarn able to locate the cspell binaries when invoked via bash like this?

Something like:

```
cspell lint **/*.mdx
```

Comment (ptgott) [reply package.json:7]: By default, cspell looks for a configuration in the current directory. I wanted to make it possible for docs authors to edit the `cspell.json` in a particular `gravitational/teleport` submodule

Comment (stevenGravy) [reply package.json:7]: I'd just have one that is maintained at the docs directory or base dir . We already have a starting list of words

Comment (zmb3) [reply package.json:7]: It would be best if it's in the Teleport repo, as that's where the content is, and it's most efficient to be able to add new allowed words in the same PR that introduces them.

I don't see why we need a different config per-branch though, a single cpell.json in teleport/docs should do, right?

Comment (ptgott) [reply package.json:7]: By teleport/docs are you referring to the `gravitational/docs` repo?

`cspell.json` includes the [list of words to ignore](https://cspell.org/configuration/). If we expect a separate config file within each `gravitational/teleport` submodule, docs authors can add words to the list in the same PR where they introduce docs edits.

It's difficult to coordinate changes that involve both `gravitational/teleport` and `gravitational/docs` without stopping the docs from building for a few hours to a day or two. Since docs pages use a specialized vocabulary, new changes are likely to introduce unsupported words, and requiring PRs in two repos would place an undue burden on docs authors.

Comment (zmb3) [reply package.json:7]: No, I'm referring to the `docs` subdirectory of the `teleport` repo.

> and requiring PRs in two repos would place an undue burden on docs authors

That's exactly what I'm trying to avoid. I'm arguing for keeping the allow list int he same place as the content (teleport).

Comment (ptgott) [reply package.json:7]: Just to make sure I understand, how would the `spellcheck` command change if we were to put the `cspell.json` file in `gravitational/teleport/docs`? 

As far as I can tell, we would change it to:

```json
"spellcheck": "find content -mindepth 1 -maxdepth 1 -type d | xargs -I {} bash -c 'cspell lint --no-progress --config {}/docs/cspell.json {}/docs/pages/**/*.mdx'", 
```
I.e., we'd make the config path `{}/docs/cspell.json` instead of `{}/cspell.json`. 

This would be because `cspell.json` would be in the `docs` subdirectory of the `teleport` repo, which maps to each first-level subdirectory within `gravitational/docs/content` via git submodules. E.g., there would be a `cspell.json` in `content/12.x/docs`, `content/13.x/docs`, etc.

Comment (zmb3) [reply package.json:7]: Ah, right - that makes sense. Thanks for clarifying. This approach sounds good to me.

**Review by hatched — APPROVED** [review]
Looks good! Can you add default config files to the existing branches so that it's ready to go for someone to modify later on?

**Comment by ptgott** [comment]
@hatched Thanks! I'll make sure we include the default config files in the `teleport` submodules before we merge this.

On another note, I'm trying to think of a way to make it clearer to users that they can add words to the ignore list in `cspell.json`. Currently, the output looks like this on my workstation:

```
 % find content -mindepth 1 -maxdepth 1 -type d | xargs -I {} bash -c 'cspell lint --no-progress --config {}/cspell.json {}/docs/pages/**/*.mdx'
/Users/paulgottschling/Documents/docs/content/12.x/docs/pages/management/join-services-to-your-cluster.mdx:3:77 - Unknown word (Telepoert)
CSpell: Files checked: 139, Issues found: 1 in 1 files
CSpell: Files checked: 125, Issues found: 0 in 0 files
/Users/paulgottschling/Documents/docs/content/10.x/docs/pages/reference/backends.mdx:704:42 - Unknown word (CREATEDB)
/Users/paulgottschling/Documents/docs/content/10.x/docs/pages/reference/backends.mdx:710:31 - Unknown word (CREATEDB)
/Users/paulgottschling/Documents/docs/content/10.x/docs/pages/reference/backends.mdx:713:30 - Unknown word (CREATEDB)
/Users/paulgottschling/Documents/docs/content/10.x/docs/pages/reference/config.mdx:658:45 - Unknown word (HOTP)
CSpell: Files checked: 109, Issues found: 4 in 2 files
/Users/paulgottschling/Documents/docs/content/13.x/docs/pages/reference/terraform-provider.mdx:229:260 - Unknown word (doesn)
/Users/paulgottschling/Documents/docs/content/13.x/docs/pages/reference/terraform-provider.mdx:591:6 - Unknown word (flexi)
/Users/paulgottschling/Documents/docs/content/13.x/docs/pages/reference/terraform-provider.mdx:591:43 - Unknown word (Flexi)
/Users/paulgottschling/Documents/docs/content/13.x/docs/pages/reference/terraform-provider.mdx:969:102 - Unknown word (indentifies)
/Users/paulgottschling/Documents/docs/content/13.x/docs/pages/reference/terraform-provider.mdx:1061:89 - Unknown word (databaes)
/Users/paulgottschling/Documents/docs/content/13.x/docs/pages/reference/terraform-provider.mdx:1214:89 - Unknown word (databaes)
/Users/paulgottschling/Documents/docs/content/13.x/docs/pages/reference/terraform-provider.mdx:1380:103 - Unknown word (stategy)
/Users/paulgottschling/Documents/docs/content/13.x/docs/pages/reference/terraform-provider.mdx:1396:51 - Unknown word (valueg)
CSpell: Files checked: 139, Issues found: 8 in 1 files
```

Would it make sense to:
- Move the `spellcheck` command into a shell script
- If an individual `cspell` command fails, print a message indicating that the user should either fix the spelling or add a word to `cspell.json` in the appropriate `teleport` branch?

**Comment by hatched** [comment]
> * Move the `spellcheck` command into a shell script

Doing this does give a lot more room for documentation and customization than trying to cram it into one line but I'll leave that up to you. 

> * If an individual `cspell` command fails, print a message indicating that the user should either fix the spelling or add a word to `cspell.json` in the appropriate `teleport` branch?

This is a great idea as it would probably pre-empt the inevitable questions in the docs channel "The linter is saying X is not a word, how do I tell it that it is?"  😁

File: check-spelling.sh
Comment (hatched) [inline check-spelling.sh:1]: This should probably be in the `scripts/` folder.

**Comment by ptgott** [comment]
To avoid breaking docs builds when merging this PR, we'll need to merge this `gravitational/teleport` PR and all of its backports first: https://github.com/gravitational/teleport/pull/23783

**Review by hatched — DISMISSED** [review]

**Comment by ptgott** [comment]
@hatched Looks like my last spelling fix voided your review 🙃 . Would you mind stamping again? Thanks!

**Review by stevenGravy — APPROVED** [review]

**Pull request merged**
